### PR TITLE
feat: add dest HTTP compression

### PR DESCRIPTION
**Pull request by mstopa-splunk**
solves https://github.com/splunk/splunk-connect-for-syslog/issues/2144

### CODE REVIEW

**Review by ikheifets-splunk — APPROVED** [review]

File: docs/destinations.md [hunk 1]
```diff
@@ -144,6 +144,19 @@ can be supported (defined by syslog-ng OSE)
 | SC4S_DEST_SPLUNK_HEC_&lt;ID&gt;_MODE | string | "GLOBAL" or "SELECT" |
 | SC4S_DEST_SPLUNK_HEC_DEFAULT_TLS_VERIFY | yes(default) or no | verify HTTP(s) certificate |
 
+### HTTP Compression
+
+HTTP traffic compression allows reducing the network connection bandwidth.\
+The currently supported version of `syslog-ng` offers two out of four compression algorithms from the `curl` library: `deflate` and `gzip`. Out of these two, only `gzip` has been implemented in SC4S for compressing transmitted traffic, because Splunk does not support `deflate`.\
+Utilizing compression may result in lower CPU load and increased utilization of RAM. It may also cause a decrease in performance. In tests, we observed a decrease in message processing speed by 6-7%.\
```
Comment (jenworthington) [inline docs/destinations.md:151]: Using the 'gzip' compression algorithm can result in lower CPU load and increased utilization of RAM. The algorithm may also cause a decrease in performance. Tests observed a decrease in message processing speed by 6% to 7%.\

File: docs/destinations.md [hunk 2]
```diff
@@ -144,6 +144,19 @@ can be supported (defined by syslog-ng OSE)
...
+Compression affects the content but not the HTTP headers. Enabling batch packet processing will make the solution particularly efficient as it allows for compressing a larger number of logs at once.
```
Comment (jenworthington) [inline docs/destinations.md:152]: Compression affects the content but does not affect the HTTP headers. Enable batch packet processing to make the solution particularly efficient, as this allows compression of a large number of logs at once.

File: docs/destinations.md [hunk 3]
```diff
@@ -144,6 +144,19 @@ can be supported (defined by syslog-ng OSE)
...
+
+
+| Variable | Values        | Description |
+|----------|---------------|-------------|
+| SC4S_DEST_SPLUNK_HEC_&lt;ID&gt;_HTTP_COMPRESSION;       | yes or no(default) | compress outgoing HTTP traffic using gzip method |
```
Comment (jenworthington) [inline docs/destinations.md:157]: Compresses outgoing HTTP traffic using the 'gzip' compression algorithm. |

Comment (jenworthington) [reply docs/destinations.md:157]: Yes or no (default).

File: docs/destinations.md [hunk 4]
```diff
@@ -144,6 +144,19 @@ can be supported (defined by syslog-ng OSE)
...
+
+
 ## Syslog Standard destination.
 
 Note: in many cases destinations incorrectly assert "syslog" support. IETF standards RFC5424, RFC5425, RFC6587 define the use of "syslog" as a network protocol. Often the actual configuration required is Legacy BSD syslog which is NOT a standard and was documented "historically" in RFC3164 see BSD Destination section.
```
Comment (jenworthington) [inline docs/destinations.md:162]: Note: in many cases, destinations incorrectly assert "syslog" support. Internet Engineering Task Force standards RFC5424, RFC5425, and RFC6587 define the use of "syslog" as a network protocol. Often the actual configuration required is Legacy BSD syslog which is not a standard and was documented in RFC3164.

File: docs/destinations.md [hunk 5]
```diff
@@ -144,6 +144,19 @@ can be supported (defined by syslog-ng OSE)
 | SC4S_DEST_SPLUNK_HEC_&lt;ID&gt;_MODE | string | "GLOBAL" or "SELECT" |
 | SC4S_DEST_SPLUNK_HEC_DEFAULT_TLS_VERIFY | yes(default) or no | verify HTTP(s) certificate |
 
+### HTTP Compression
+
```
[the remark this answers is not in the document]
Comment (mstopa-splunk) [reply docs/destinations.md:148]: done

File: docs/destinations.md [hunk 6]
```diff
@@ -144,6 +144,19 @@ can be supported (defined by syslog-ng OSE)
...
+HTTP traffic compression allows reducing the network connection bandwidth.\
+The currently supported version of `syslog-ng` offers two out of four compression algorithms from the `curl` library: `deflate` and `gzip`. Out of these two, only `gzip` has been implemented in SC4S for compressing transmitted traffic, because Splunk does not support `deflate`.\
```
[the remark this answers is not in the document]
Comment (mstopa-splunk) [reply docs/destinations.md:150]: done

**Pull request merged**